README.md

```diff
@@ -1,6 +1,39 @@
-# Sourcegraph Bitbucket Server Integration
+# Sourcegraph for Bitbucket
 
-Sourcegraph's Bitbucket Server integration adds code intelligence to Bitbucket Server code views and pull requests.
+The Sourcegraph plugin for Bitbucket Server communicates with your Sourcegraph instance to add code intelligence to your Bitbucket Server code views and pull requests.
+
+## Installation and Usage
+
+### Prerequisites
+
+1. You must have a self-hosted Sourcegraph instance set up, v3.6 or higher.
+2. Your Sourcegraph instance should have a Bitbucket Server [external service](https://docs.sourcegraph.com/admin/external_service/bitbucket_server) configured.
+3. The `corsOrigin` property should be set in [site configuration](https://docs.sourcegraph.com/admin/config/site_config) to include the URL of your Bitbucket Server instance. Example value:
+
+```json
+    "corsOrigin": "https://bitbucket.internal.org"
+```
+
+### Configuration
+
+After installing the Sourcegraph for Bitbucket, you should configure it to point to your Sourcegraph instance.
+
+1. On Bitbucket, go to the **Administration** page
+2. Find the **Sourcegraph** entry under **Add-ons**:
+
+![Add-ons](img/add-ons.png)
+
+3. On the **Sourcegraph Settings** page, set the Sourcegraph URL to the URL of your self-hosted Sourcegraph instance:
+
+![Sourcegraph settings](img/sourcegraph-settings.png)
+
+### Usage
+
+Once configured, Sourcegraph for Bitbucket will add code intelligence hovers to code views and pull requests for all users that are logged in to your self-hosted Sourcegraph instance:
```

Comment:
Perhaps mention what would happen if a user is not logged in to the Sourcegraph instance to pre-empt the question for admins choosing to set up for their company.